### PR TITLE
Stories feed widget

### PR DESCRIPTION
**Pull request by mirabelrice**

### CODE REVIEW

File: lib/assets/javascripts/chapman_stories_feed.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  // passed in so we can ensure the value of it is truly undefined. In ES5, undefined
+  // can no longer be modified.
+
+  // window and document are passed through as local variables rather than global
+  // as this (slightly) quickens the resolution process and can be more efficiently
+  // minified (especially when both are regularly referenced in your plugin).
+
+  // Create the defaults once
+  var defaults = {
+    baseUrl: 'http://social.localchapman.edu:3000/',
```
Comment (mirabelrice) [inline lib/assets/javascripts/chapman_stories_feed.js]: I will change this back to social.chapman.edu once inside is deployed

File: lib/assets/javascripts/chapman_stories_feed.js [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+      return $outerContainer;
+    },
+
+    buildTopStoryElement: function() {
+      var classes = 'feed-column top-story';
+      var loadingHtml = '<p>Loading top story...</p>'
+      var $topStoryElement = $('<div />').addClass(classes).html(loadingHtml);
+      
+      //if top story post is not specified, default to the top story from the specified blog feed
+      var params = this.topStoryUrl ? { url: this.topStoryUrl } : { source: this.feedUrl };
```
Comment (mirabelrice) [inline lib/assets/javascripts/chapman_stories_feed.js]: Here is the logic for the top story default

Comment (tatwell) [inline lib/assets/javascripts/chapman_stories_feed.js]: Send both and let the backend (Inside) figure it out. That way, if we decide we want to change the logic or output, we can do it there and not worry about coordinating with this.

File: lib/assets/javascripts/chapman_stories_feed.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+      this.buildOutWidget();
+    },
+
+    //this is a temporary fix to remove trailing slash from blog site urls(that is how they are stored on inside db)
+    //Ideally, inside query should be able to match urls with or without trailing slash
+    parseFeedUrl: function() {
+      var parsedUrl = this.$element.data('chapmanStoriesFeedUrl') || this.settings.dataFeedUrl;
+      parsedUrl = parsedUrl.replace(/\/?$/, '');
+      return parsedUrl;
+    },
```
Comment (tatwell) [inline lib/assets/javascripts/chapman_stories_feed.js]: This isn't used, is it? I think this is a vestige of previous version. If so, please remove.

Comment (mirabelrice) [inline lib/assets/javascripts/chapman_stories_feed.js]: Yes this is being used until the query can be fixed in ElasticSearch

File: lib/assets/javascripts/chapman_stories_feed.js (hunk 2 above)
Comment (mirabelrice) [inline lib/assets/javascripts/chapman_stories_feed.js]: okay, fixed

File: app/assets/stylesheets/widgets/single_column/chapman_stories_feed.scss [hunk 4]
```diff
@@ -0,0 +1,151 @@
+.chapman-stories-feed-widget {
+	padding: $one-column-widget-spacing 0;
+  text-align: center;
```
Comment (mirabelrice) [inline app/assets/stylesheets/widgets/single_column/chapman_stories_feed.scss]: Yes my editor is using 2 spaces so i'm not sure why the formatting is off here.

**Comment by tatwell** [comment]
👍

**Pull request merged**